File: MATTHEW/leetcode/AIcoding/1.Gemini-2.5/java_solutions/3074-apple-redistribution-into-boxes.java
import java.util.Arrays;
import java.util.Collections;

class Solution {
    public int minimumBoxes(int[] apple, int[] capacity) {
        int totalApples = 0;
        for (int a : apple) {
            totalApples += a;
        }

        Integer[] capacityObj = new Integer[capacity.length];
        for (int i = 0; i < capacity.length; i++) {
            capacityObj[i] = capacity[i];
        }

        Arrays.sort(capacityObj, Collections.reverseOrder());

        int boxesUsed = 0;
        int currentCapacity = 0;
        for (int cap : capacityObj) {
            currentCapacity += cap;
            boxesUsed++;
            if (currentCapacity >= totalApples) {
                break;
            }
        }
        return boxesUsed;
    }
}

File: MATTHEW/leetcode/AIcoding/5.Qwen-3/cpp_solutions/3074-apple-redistribution-into-boxes.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int minimumBoxes(vector<int>& apple, vector<int>& capacity) {
        sort(apple.begin(), apple.end());
        sort(capacity.begin(), capacity.end());

        int boxes = 0;
        int i = apple.size() - 1;
        int j = capacity.size() - 1;

        while (i >= 0) {
            if (capacity[j] >= apple[i]) {
                boxes++;
                i--;
                j--;
            } else {
                boxes++;
                apple[i] -= capacity[j];
                j--;
            }
        }

        return boxes;
    }
};